scripts/run_test262_utils.js

```diff
@@ -0,0 +1,223 @@
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+
+const parse = require("..").parse;
+
+const modulePattern = /^\s*-\s*module\s*$|^\s*flags\s*:.*\bmodule\b/m;
+const noStrictPattern = /^\s*-\s*noStrict\s*$|^\s*flags\s*:.*\bnoStrict\b/m;
+const onlyStrictPattern = /^\s*-\s*onlyStrict\s*$|^\s*flags\s*:.*\bonlyStrict\b/m;
+const rawPattern = /^\s*-\s*raw\s*$|^\s*flags\s*:.*\braw\b/m;
+const testNamePattern = /^(?!.*_FIXTURE).*\.[jJ][sS]$/;
+
+function flatten(array) {
+  const flattened = [];
+  array.forEach(function(element) {
+    if (Array.isArray(element)) {
+      flattened.push.apply(flattened, element);
+    } else {
+      flattened.push(element);
+    }
+  });
+  return flattened;
+}
+
+function hasEarlyError(src) {
+  return !!(
+    src.match(/^\s*negative:\s*$/m) && src.match(/^\s+phase:\s*early\s*$/m)
+  );
+}
+
+function readDir(dirName) {
+  return new Promise(function(resolve, reject) {
+    fs.readdir(dirName, function(err, contents) {
```

Comment:
Any objections to adding a devDependency on [`util.promisify`](https://www.npmjs.com/package/util.promisify)?

It's available natively in Node >= 8, the package just shims it if it's missing.